entity-service/src/main/kotlin/com/egm/stellio/entity/web/EntityHandler.kt

```diff
@@ -119,72 +132,79 @@ class EntityHandler(
             }
     }
 
     /**
      * Implements 6.6.3.2 - Update Entity Attributes
      *
      */
-    fun updateEntityAttributes(req: ServerRequest): Mono<ServerResponse> {
-        val uri = req.pathVariable("entityId")
-        val contextLink = extractContextFromLinkHeader(req)
-
-        return req.bodyToMono<String>()
+    @PatchMapping("/{entityId}/attrs", consumes = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE, JSON_MERGE_PATCH_CONTENT_TYPE])
+    fun updateEntityAttributes(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @PathVariable entityId: String,
+        @RequestBody body: Mono<String>
+    ): Mono<ResponseEntity<*>> {
+        val contextLink = extractContextFromLinkHeader(httpHeaders.getOrEmpty("Link"))
+
+        return body
             .doOnNext {
-                if (!entityService.exists(uri)) throw ResourceNotFoundException("Entity $uri does not exist")
+                if (!entityService.exists(entityId)) throw ResourceNotFoundException("Entity $entityId does not exist")
             }
             .map {
-                entityService.updateEntityAttributes(uri, it, contextLink)
+                entityService.updateEntityAttributes(entityId, it, contextLink)
             }
-            .flatMap {
-                logger.debug("Update $it attributes on entity $uri")
+            .map {
+                logger.debug("Update $it attributes on entity $entityId")
                 if (it.notUpdated.isEmpty())
-                    status(HttpStatus.NO_CONTENT).build()
+                    ResponseEntity.status(HttpStatus.NO_CONTENT).build<String>()
                 else
-                    status(HttpStatus.MULTI_STATUS).body(BodyInserters.fromValue(it))
+                    ResponseEntity.status(HttpStatus.MULTI_STATUS).body(it)
             }
     }
 
     /**
      * Implements 6.7.3.1 - Partial Attribute Update
      * Current implementation is basic and only update the value of a property.
      */
-    fun partialAttributeUpdate(req: ServerRequest): Mono<ServerResponse> {
-        val attr = req.pathVariable("attrId")
-        val uri = req.pathVariable("entityId")
-        val contextLink = extractContextFromLinkHeader(req)
-
-        return req.bodyToMono<String>()
+    @PatchMapping("/{entityId}/attrs/{attrId}", consumes = [MediaType.APPLICATION_JSON_VALUE, JSON_LD_CONTENT_TYPE, JSON_MERGE_PATCH_CONTENT_TYPE])
+    fun partialAttributeUpdate(
+        @RequestHeader httpHeaders: HttpHeaders,
+        @PathVariable entityId: String,
+        @PathVariable attrId: String,
+        @RequestBody body: Mono<String>
+    ): Mono<ResponseEntity<*>> {
+        val contextLink = extractContextFromLinkHeader(httpHeaders.getOrEmpty("Link"))
+
+        return body
             .map {
-                entityService.updateEntityAttribute(uri, attr, it, contextLink)
+                entityService.updateEntityAttribute(entityId, attrId, it, contextLink)

```

Comment:
It is not related to this PR but it is not consistent that validation is done on entityId given but not there.